runtime/mount/agent.py

```diff
@@ -2843,6 +2874,36 @@ async def handle_stream(self, *, use_beta_api: bool, **kwargs) -> tuple[Message
 
             raise
 
+    async def _summarize_thinking(self, thinking_text: str, is_final: bool = False) -> str:
+        try:
+            if is_final:
+                prompt = f"Summarize the reasoning process in a concise past-tense sentence (2-6 words). Focus on the key decisions or conclusions reached. Examples: 'Analyzed error logs and identified the root cause', 'Refined the search query to improve results'.\n\n{thinking_text}"
+            else:
+                prompt = f"Summarize the most recent thoughts in this reasoning process into a brief, active phrase (2-6 words). Focus on what is currently being pondered or analyzed. Examples: 'Evaluating error handling', 'Checking authentication logic', 'Refining the search query'.\n\n{thinking_text}"
+
+            summary_msg = await self.client.messages.create(
+                model="claude-3-haiku-20240307",
```

Comment:
what is the rationale behind using such an old model?

Reply:
idk, didn't look up what the latest was